### DISCUSSION TITLE
Installation error in HPC

### DISCUSSION THREAD

**Discussion by gmaurya20** [idaholab/moose#21001] [Q&A Getting Started] [answered]
I have been trying to install MOOSE in HPC. I am using an offline installation instruction given on the MOOSEframework website. These are the version of the prerequisites that I am using to install MOOSE:
Python-3.7.4
mpich-3.2
gcc/ver-9.1.0
Cmake-3.23.1

when I am running the script " **./scripts/update_and_rebuild_petsc.sh --skip-submodule-update --with-packages-download-dir=~/offline/downloads --prefix=$HOME/libs/petsc** " I am getting error. i ahve attached the screenshot of the error.

please help me to solve this error.
Thank You,
Regards 
Gaurav
![Screenshot from 2022-05-11 19-28-50](https://user-images.githubusercontent.com/96609970/167872863-a2366077-8b11-4eb8-954c-ec82878608d4.png)

**Comment by milljm** [comment]
Can you share the configure.log that the output is referencing?

**Reply by gmaurya20** [reply]
[configure.log](https://github.com/idaholab/moose/files/8671833/configure.log)

**Comment by milljm** [comment, marked as answer]
There seems to be a weird mismatch between GCC versions for `mpicc --version` vs `mpif90 --version`:

```pre
Executing: mpif90 --version
stdout:
GNU Fortran (GCC) 4.8.5 20150623 (Red Hat 4.8.5-4)
```

```pre
Executing: mpicc --version
stdout:
gcc (GCC) 9.1.0
```

These need to match up... as throughout the build process both of these locations are being included with linker flags. The error also supports this theory:

```pre
Possible ERROR while running linker: exit code 1
stderr:
/home/gmaurya20/libs/petsc/lib/libsuperlu_dist.so: undefined reference to `GOMP_loop_nonmonotonic_dynamic_next'
/home/gmaurya20/libs/petsc/lib/libsuperlu_dist.so: undefined reference to `GOMP_loop_nonmonotonic_guided_start'
/home/gmaurya20/libs/petsc/lib/libsuperlu_dist.so: undefined reference to `GOMP_loop_nonmonotonic_dynamic_start'
/home/gmaurya20/libs/petsc/lib/libsuperlu_dist.so: undefined reference to `GOMP_loop_nonmonotonic_guided_next'
/home/gmaurya20/libs/petsc/lib/libsuperlu_dist.so: undefined reference to `GOMP_taskloop'
```

When I do a search for "GOMP_loop_nonmonotonic_dynamic_next" I run into several posts about differences between building and then running against different GCC versions. Ref:  https://gcc.gnu.org/legacy-ml/fortran/2019-05/msg00037.html

What you should do is determine if you can get the same GCC versions for each MPI wrapper:

```bash
mpicc --version
mpicxx --version
mpif90 --version
mpif77 --version
```
Each of the above commands should return the same GCC version.

**Reply by gmaurya20** [reply]
I installed a new mpich and it worked.
